orchestrator-salt/src/main/resources/salt/salt/metering/scripts/metering_package_manager.sh

```diff
@@ -0,0 +1,316 @@
+   upgrade command arguments:
+     -d, --from-date      <yyyy-mm-dd>       Install RPM if the installation date is lower than this value.
+     -f, --rpm-file       <LOCAL RPM FILE>   Local RPM file to be installed (required for upgrade only)
+     -u, --custom-rpm-url <CUSTOM RPM URL>   Custom URL that will be used instead of the default archive URL
+EOF
+}
+
+function do_exit() {
+  local code=$1
+  local message=$2
+  if [[ "$message" == "" ]]; then
+    info "Exit code: $code"
+  else
+    info "Exit code: $code, --- STATUS MESSAGE --- $message --- STATUS MESSAGE ---"
+  fi
+  exit $code
+}
+
+function init_logfile() {
+  mkdir -p $LOGFILE_FOLDER
+  local timestamp=$(date +"%Y%m%d-%H%M%S")
+  LOGFILE="$LOGFILE_FOLDER/metering-upgrade-${timestamp}.log"
+  touch $LOGFILE
+  cleanup_old_logs
+  info "The following log file will be used: $LOGFILE"
+}
+
+function init_salt_prefix() {
+  SALT_BIN_PREFIX=$(find /opt -maxdepth 1 -type d -iname "salt_*" | xargs -I{} echo "{}/bin")
+  if [[ "$SALT_BIN_PREFIX" == "" ]]; then
+    SALT_BIN_PREFIX="/opt/salt_*/bin"
+  fi
+}
+
+function cleanup_old_logs() {
+  ls -1tr $LOGFILE_FOLDER/metering-upgrade*.log | head -n -3 | xargs --no-run-if-empty rm
+}
+
+function info() {
+  log "$1"
+}
+
+function debug() {
+  log "$1" "true"
+}
+
+function log() {
+  local timestamp=$(date +"%Y-%m-%dT%H:%M:%S.%3N%z")
+  local debug=$2
+  echo "$timestamp $1" >> $LOGFILE
+  if [[ "$2" == "" ]]; then
+    echo "$1"
+  fi
+}
+
+function init_proxy_configs() {
+  local proxy_settings=$($SALT_BIN_PREFIX/salt-call pillar.items --out=json | jq .local.proxy)
+  CURL_EXTRA_PARAMS="--connect-timeout ${CURL_CONNECT_TIMEOUT}"
+  if [[ "${proxy_settings}" != "null" ]]; then
+    local proxy_protocol=$(echo "${proxy_settings}" | jq .protocol | tr -d '"')
+    local proxy_host=$(echo "${proxy_settings}" | jq .host | tr -d '"')
+    local proxy_user=$(echo "${proxy_settings}" | jq .user | tr -d '"')
+    local proxy_password=$(echo "${proxy_settings}" | jq .password | tr -d '"')
+    local no_proxy_hosts=$(echo "${proxy_settings}" | jq .noProxyHosts | tr -d '"')
+    if [[ "${proxy_host}" != "null" ]]; then
+      local proxy_url="${proxy_protocol}://${proxy_host}"
+      info "Https proxy is set: ${proxy_url}"
+      CURL_EXTRA_PARAMS="${CURL_EXTRA_PARAMS} --proxy ${proxy_url}"
+      if [[ "${proxy_user}" != "null" ]]; then
+        https_proxy="${proxy_protocol}://${proxy_user}:${proxy_password}@${proxy_host}"
+        CURL_EXTRA_PARAMS="${CURL_EXTRA_PARAMS} --proxy-user ${proxy_user}:${proxy_password}"
+      else
+        https_proxy="${proxy_url}"
+      fi
+    fi
+    if [[ "${no_proxy_hosts}" != "null" ]]; then
+      no_proxy=${no_proxy_hosts}
+      CURL_EXTRA_PARAMS="${CURL_EXTRA_PARAMS} --noproxy ${no_proxy}"
+    fi
+  else
+    info "Https proxy is not set."
+  fi
+}
+
+function rpm_upgrade() {
+  debug "Execute command: yum remove -y $METERING_PACKAGE_NAME"
+  yum remove -y $METERING_PACKAGE_NAME
```

Comment:
what happens if the remove fails? would the `rpm -i` also fail?

Reply:
then rpm -i fails, modified to use with `&&`